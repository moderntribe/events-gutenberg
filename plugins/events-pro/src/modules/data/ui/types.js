/* eslint-disable max-len */
/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_PRO_STORE } from '@moderntribe/events-pro/data/prefix';

export const TOGGLE_REPEAT_EVENTS_BLOCK_VISIBILITY = `${ PREFIX_EVENTS_PRO_STORE }/TOGGLE_REPEAT_EVENTS_BLOCK_VISIBILITY`;
