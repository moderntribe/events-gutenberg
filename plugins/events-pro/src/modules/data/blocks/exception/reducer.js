export default function exception( state = {}, action ) {
	switch ( action.type ) {
		default:
			return state;
	}
}
