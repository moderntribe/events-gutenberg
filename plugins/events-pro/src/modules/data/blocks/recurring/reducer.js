/**
 * Internal dependencies
 */
import * as types from './types';

export default function recurring( state = [], action ) {
	switch ( action.type ) {
		case types.ADD_RECURRING_FIELD:
			return [
				...state,
				action.payload,
			];
		case types.REMOVE_RECURRING_FIELD:
			return state.filter( field => field.id === action.payload.id );
		default:
			return state;
	}
}
