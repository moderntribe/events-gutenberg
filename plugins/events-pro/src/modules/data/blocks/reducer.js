/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import additionalFields from '@moderntribe/events-pro/data/blocks/additional-fields';

export default combineReducers( {
	additionalFields,
} );
