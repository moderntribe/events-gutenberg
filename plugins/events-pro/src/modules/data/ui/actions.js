/**
 * Internal dependencies
 */
import * as types from './types';

export const toggleRepeatBlocksVisibility = () => ( {
	type: types.TOGGLE_REPEAT_EVENTS_BLOCK_VISIBILITY,
} );
