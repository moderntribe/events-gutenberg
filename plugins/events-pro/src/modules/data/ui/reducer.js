/**
 * Internal dependencies
 */
import * as types from './types';

export const DEFAULT_STATE = {
	isRepeatBlockVisible: false,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.TOGGLE_REPEAT_EVENTS_BLOCK_VISIBILITY:
			return {
				...state,
				isRepeatBlockVisible: ! state.isRepeatBlockVisible,
			};
		default:
			return state;
	}
};
