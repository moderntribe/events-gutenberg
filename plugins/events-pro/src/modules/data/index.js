/**
 * External dependencies
 */
import reducer from './reducers';

import { actions } from '@moderntribe/common/data/plugins';
import { store } from '@moderntribe/common/store';
import { EVENTS_PRO_PLUGIN_NAME } from './constants';

export const initStore = () => {
	const { dispatch, injectReducers } = store;

	dispatch( actions.addPlugin( EVENTS_PRO_PLUGIN_NAME ) );
	injectReducers( { [ EVENTS_PRO_PLUGIN_NAME ]: reducer } );
};

export const getStore = () => store;
