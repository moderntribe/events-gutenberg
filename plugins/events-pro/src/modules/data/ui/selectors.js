/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { constants } from '@moderntribe/common/data/plugins';

export const getUI = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ].ui;

export const isRepeatBlockVisible = createSelector(
	[ getUI ],
	( ui ) => ui.isRepeatBlockVisible
);
