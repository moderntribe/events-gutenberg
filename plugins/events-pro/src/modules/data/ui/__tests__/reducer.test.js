/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/events-pro/data/ui';
import reducer, { DEFAULT_STATE } from '@moderntribe/events-pro/data/ui/reducer';

describe( '[STORE] - UI reducer', () => {
	it( 'Should return the default state', () => {
		expect( reducer( undefined, {} ) ).toEqual( DEFAULT_STATE );
	} );

	it( 'Should toggle repeat panel', () => {
		const current = reducer( { isRepeatBlockVisible: false }, actions.toggleRepeatBlocksVisibility() );
		expect( current ).toMatchSnapshot();
	} );
} );
