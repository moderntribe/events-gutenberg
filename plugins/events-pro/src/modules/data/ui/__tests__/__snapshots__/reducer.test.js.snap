// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - UI reducer Should toggle repeat panel 1`] = `
Object {
  "isRepeatBlockVisible": true,
}
`;
